epsagon/events/requests.py

```diff
@@ -65,15 +69,42 @@ def __init__(self, wrapped, instance, args, kwargs, start_time, response,
         )
 
         if response is not None:
-            self.update_response(response)
+            self.update_response(response, kwargs.get('stream', False))
 
         if exception is not None:
             self.set_exception(exception, traceback.format_exc())
 
-    def update_response(self, response):
+    @staticmethod
+    def _get_response_body(response, is_stream):
+        """
+        Gets the response body from the response
+        :param response: the Response object
+        :param is_stream: the param value as given to the original request
+        :return: the response body, None on failure
+        """
+        try:
+            if is_stream:
+                data = response.raw.peek()
+            else:
+                data = response.content
+        except Exception: # pylint: disable=broad-except
+            return None
+
+        if not data:
+            return data
+        try:
+            return json.loads(data)
+        except ValueError:
+            if isinstance(data, bytes):
+                try:
+                    return data.decode('utf-8')
+                except UnicodeDecodeError:
+                    return str(data)
```

Comment:
I would change this to a single return data at the end, then assign data to its value instead of returning at each step